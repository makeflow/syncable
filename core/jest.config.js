module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',

  clearMocks: true,

  coverageDirectory: 'coverage',

  globals: {
    'ts-jest': {
      tsConfig: 'src/test/tsconfig.test.json',
    },
  },
};
