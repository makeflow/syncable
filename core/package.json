{
  "name": "@syncable/core",
  "version": "3.0.0-wip.22",
  "publishConfig": {
    "access": "public"
  },
  "description": "Syncable.",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/makeflow/syncable.git"
  },
  "keywords": [
    "syncable"
  ],
  "author": "Chengdu Mufan Technology Co., Ltd.",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/makeflow/syncable/issues"
  },
  "homepage": "https://github.com/makeflow/syncable#readme",
  "scripts": {
    "lint:library": "tslint --project src/library",
    "lint": "yarn lint:library"
  },
  "main": "bld/library/index.js",
  "types": "bld/library/index.d.ts",
  "files": [
    "src/**/*.ts",
    "bld"
  ],
  "peerDependencies": {
    "mobx": "5",
    "rxjs": "6"
  },
  "dependencies": {
    "deep-diff": "^1.0.2",
    "extendable-error": "^0.1.5",
    "lodash": "^4.17.14",
    "replace-object": "^0.2.1",
    "tslang": "^0.1.13",
    "uuid": "^3.3.2",
    "villa": "^0.3.1"
  },
  "devDependencies": {
    "@types/deep-diff": "^1.0.0",
    "@types/lodash": "^4.14.120",
    "@types/uuid": "^3.4.3"
  }
}
