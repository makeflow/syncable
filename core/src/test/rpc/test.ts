import {RPCPeer} from '@syncable/core';

import {createAdapterPair} from './@adapter';
import {
  ThatDefinition,
  ThisDefinition,
  thatFunctionDict,
  thisFunctionDict,
} from './@peer';

const [thisAdapter, thatAdapter] = createAdapterPair();

const thisPeer = new RPCPeer<ThisDefinition, ThatDefinition>(
  thisAdapter,
  thisFunctionDict,
);

const thatPeer = new RPCPeer<ThatDefinition, ThisDefinition>(
  thatAdapter,
  thatFunctionDict,
);

test('should call `this` remote function from `that` and get return value', async () => {
  let fooValue = await thatPeer.call('foo', 'hello, world', 5);
  // tslint:disable-next-line:no-void-expression
  let barValue = await thatPeer.call('bar', false);

  expect(fooValue).toBe('hello');
  expect(barValue).toBeUndefined();
});

test('should call `this` remote function from `that` and get error', async () => {
  await expect(thatPeer.call('bar', true)).rejects.toMatchInlineSnapshot(
    `[RPCError: Bar error occurred]`,
  );
});

test('should call `that` remote function from `this` and get return value', async () => {
  let value = await thisPeer.call('yoha', 'hello, world', 5);

  expect(value).toBe('hello');
});
