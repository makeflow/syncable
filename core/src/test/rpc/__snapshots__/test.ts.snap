// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call \`this\` remote function from \`that\` and get error 1`] = `[RPCError: Bar error occurred]`;
